docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,514 @@
+
+```json
+{
+    "title": "MyPage Title",
+    "meta": [
+        {
+            "name": "description",
+            "content":"MyPage Description",
+        },
+        {
+            "property": "og:title",
+            "content": "MyPage Title",
+        },
+        {
+            "property": "og:image",
+            "content": {
+                "id": "6fU8dkL1P9eZlPE9JPw89n"
+            }
+        }
+    ],
+}
+```
+
+- Add a SEO Assets field (many file, media input) to the model.
+- Create a link field type in the models you need SEO and link it to the model you just created.
+
+The resulting content model should look like this.
+
+![SEO Content Model](../../static/img/SEO%20content%20model.png)
+
+In order to obtain the `id` used for meta tags whose content is an asset, you need to select the asset entry and, on the right side panel of the asset, select **Info** and copy the **id**. Later in the application you'll use this **id** to generate the URL for the asset.
+
+On the application itself, the steps to customize the SEO are the following:
+
+- Define a _default_ SEO data.
+- Fetch page data as you normally would.
+- Obtain the SEO data from the Contentful page data (example code for the Contentful SEO _parser_)
+
+```js
+import { mapValues } from 'lodash';
+
+// This function will determine if a meta entry's content is an id
+// If it is, it obtains the url to the asset that matches the id
+// Otherwise it returns the content 
+
+const getContent = (content, assets) => {
+    if (isPlainObject(content)) {
+        const asset = assets.find(({ sys }) => sys.id === content.id);
+
+        return asset.fields.file.url.slice(2);
+    }
+
+    return content;
+};
+
+// Provided with the result from the Contentful API call and the entryID for the current page content,
+// This function will extract the SEO data, and complete it with the URL of any needed asset.
+
+const contentfulSEOParser = (contentfulData, entryID) => {
+    const entry = contentfulData.items.find((item) => item.sys.id === entryID);
+    const seoID = entry.fields.seo.id;
+    const seoEntry = entry.includes.Entry.find((entry) => entry.sys.id === seoID);
+    let seoContent;
+
+    try {
+        seoContent = JSON.parse(seoEntry.fields.seo);
+    } catch (parsingError) {
+        console.err('Unvalid JSON in SEO content');
+        console.err(parsingError);
+        return;
+    }
+
+    // Convert every content: { id: ''} into a content: url, keep unchanged otherwise
+    const content = {
+        title: seoContent.title,
+        meta: seoContent.meta.map((entry) => mapValues(entry, (value) => getContent(value, contentfulData.includes.Asset)))
+        link: seoContent.link.map((entry) => mapValues(entry, (value) => getContent(value, contentfulData.includes.Asset)))
+    }
+
+    return content;
+}
+```
+
+- Merge the _default_ SEO data with the SEO data fetched from the Contentful.
```

Comment:
Is this necessary? The `Seo` component already does this.

Reply:
The `SEO` component only accepts one object with SEO data. If you have local SEO data and SEO data from Contentful, you need to combine it in the object you will provide to the `SEO` component.

One could argue the defaults could be written directly to `Head` in the `App` component, right before the `SEO` component is rendered, but this way you wouldn't be taking advantage of the tag de-duplication behavior that the `SEO` component provides.

Reply:
The SEO component uses the `Head` component internally. If we use it with the defaults in the `App` component, only the new ones set by the page will be overridden. The previous default ones that are not overridden will still be available. So there is no need to do this I think 🤔